includes/class-access-token.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * File containing the class Access_Token.
+ *
+ * @package wp-job-manager
+ */
+
+namespace WP_Job_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * A token that provides access to a private resource.
+ *
+ * @since $$next-version$$
+ */
+class Access_Token {
+
+	/**
+	 * An expiry timestamp.
+	 *
+	 * @var int|null
+	 */
+	private ?int $expiry;
+
+	/**
+	 * Token's payload. It is hashed together with the token.
+	 *
+	 * @var array
+	 */
+	private array $payload;
```

Comment:
It's confusing how 'payload' is used for multiple things, being an instance property and then loaded from the database as something else. Maybe it's simpler to make this into just `$user_id`?

Reply:
So the idea is that the payload can be many different things that relate to what this token authorize you to do. As we store the hashed payload in the db, it is a good idea to have it as generic as it can get for the API to be backwards compatible in the future in case we want to hash more stuff.

I was thinking also that maybe we should migrate the existing unsubscribe tokens to use this new API. These links already use alert_id and user_id as part of the token so just having a $user_id isn't enough. Also, sometimes this is going to be the $user_id but for accountless alerts it is going to be a $post_id.

I agree with your point though that it's confusing and even payload isn't a good enough name. I'll look into moving around things to make it more clear.